### PR TITLE
Feature/app fabric app authorization

### PR DESCRIPTION
**Pull request by markoskandylis**
### Description

This PR adds a resource for aws_appfabric_app_authorization

https://docs.aws.amazon.com/appfabric/latest/api/API_CreateAppAuthorization.html

### Relations

Relates #34549.
Requires https://github.com/hashicorp/terraform-provider-aws/pull/37468.

### Output from Acceptance Testing

```console
make testacc TESTARGS='-run=TestAccAppFabric_serial/AppAuthorization' PKG=appfabric  ACCTEST_PARALLELISM=1
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go1.22.2 test ./internal/service/appfabric/... -v -count 1 -parallel 1  -run=TestAccAppFabric_serial/AppAuthorization -timeout 360m
=== RUN   TestAccAppFabric_serial
=== PAUSE TestAccAppFabric_serial
=== CONT  TestAccAppFabric_serial
=== RUN   TestAccAppFabric_serial/AppAuthorization
=== RUN   TestAccAppFabric_serial/AppAuthorization/tags
=== PAUSE TestAccAppFabric_serial/AppAuthorization/tags
=== RUN   TestAccAppFabric_serial/AppAuthorization/basic
=== PAUSE TestAccAppFabric_serial/AppAuthorization/basic
=== RUN   TestAccAppFabric_serial/AppAuthorization/disappears
=== PAUSE TestAccAppFabric_serial/AppAuthorization/disappears
=== RUN   TestAccAppFabric_serial/AppAuthorization/apiKeyUpdate
=== PAUSE TestAccAppFabric_serial/AppAuthorization/apiKeyUpdate
=== RUN   TestAccAppFabric_serial/AppAuthorization/oath2Update
=== PAUSE TestAccAppFabric_serial/AppAuthorization/oath2Update
=== CONT  TestAccAppFabric_serial/AppAuthorization/tags
=== CONT  TestAccAppFabric_serial/AppAuthorization/apiKeyUpdate
=== CONT  TestAccAppFabric_serial/AppAuthorization/oath2Update
=== CONT  TestAccAppFabric_serial/AppAuthorization/disappears
=== CONT  TestAccAppFabric_serial/AppAuthorization/basic
--- PASS: TestAccAppFabric_serial (98.97s)
    --- PASS: TestAccAppFabric_serial/AppAuthorization (0.00s)
        --- PASS: TestAccAppFabric_serial/AppAuthorization/tags (32.37s)
        --- PASS: TestAccAppFabric_serial/AppAuthorization/apiKeyUpdate (22.05s)
        --- PASS: TestAccAppFabric_serial/AppAuthorization/oath2Update (22.09s)
        --- PASS: TestAccAppFabric_serial/AppAuthorization/disappears (10.40s)
        --- PASS: TestAccAppFabric_serial/AppAuthorization/basic (12.05s)
PASS
ok  	github.com/hashicorp/terraform-provider-aws/internal/service/appfabric	104.407s

...
```

### CODE REVIEW

**Comment by markoskandylis** [comment]
After fixing the tests:
make testacc TESTARGS='-run=TestAccAppFabric_serial/AppAuthorization' PKG=appfabric  ACCTEST_PARALLELISM=1
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go1.22.2 test ./internal/service/appfabric/... -v -count 1 -parallel 1  -run=TestAccAppFabric_serial/AppAuthorization -timeout 360m
=== RUN   TestAccAppFabric_serial
=== PAUSE TestAccAppFabric_serial
=== CONT  TestAccAppFabric_serial
=== RUN   TestAccAppFabric_serial/AppAuthorization
=== RUN   TestAccAppFabric_serial/AppAuthorization/disappears
=== PAUSE TestAccAppFabric_serial/AppAuthorization/disappears
=== RUN   TestAccAppFabric_serial/AppAuthorization/apiKeyUpdate
=== PAUSE TestAccAppFabric_serial/AppAuthorization/apiKeyUpdate
=== RUN   TestAccAppFabric_serial/AppAuthorization/oath2Update
=== PAUSE TestAccAppFabric_serial/AppAuthorization/oath2Update
=== RUN   TestAccAppFabric_serial/AppAuthorization/tags
=== PAUSE TestAccAppFabric_serial/AppAuthorization/tags
=== RUN   TestAccAppFabric_serial/AppAuthorization/basic
=== PAUSE TestAccAppFabric_serial/AppAuthorization/basic
=== CONT  TestAccAppFabric_serial/AppAuthorization/disappears
=== CONT  TestAccAppFabric_serial/AppAuthorization/tags
=== CONT  TestAccAppFabric_serial/AppAuthorization/basic
=== CONT  TestAccAppFabric_serial/AppAuthorization/oath2Update
=== CONT  TestAccAppFabric_serial/AppAuthorization/apiKeyUpdate
--- PASS: TestAccAppFabric_serial (97.39s)
    --- PASS: TestAccAppFabric_serial/AppAuthorization (0.00s)
        --- PASS: TestAccAppFabric_serial/AppAuthorization/disappears (10.64s)
        --- PASS: TestAccAppFabric_serial/AppAuthorization/tags (31.34s)
        --- PASS: TestAccAppFabric_serial/AppAuthorization/basic (11.82s)
        --- PASS: TestAccAppFabric_serial/AppAuthorization/oath2Update (21.69s)
        --- PASS: TestAccAppFabric_serial/AppAuthorization/apiKeyUpdate (21.90s)
PASS
ok  	github.com/hashicorp/terraform-provider-aws/internal/service/appfabric	102.808s

File: internal/service/appfabric/app_authorization.go [hunk 1]
```diff
@@ -0,0 +1,617 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package appfabric
+
```
Comment (meetreks) [inline internal/service/appfabric/app_authorization.go:5]: Looks Good

File: internal/service/appfabric/app_authorization.go [hunk 2]
```diff
@@ -0,0 +1,617 @@
...
+}
+
+func (r *appAuthorizationResource) Metadata(_ context.Context, request resource.MetadataRequest, response *resource.MetadataResponse) {
+	response.TypeName = "aws_appfabric_app_authorization"
+}
+
+func (r *appAuthorizationResource) Schema(ctx context.Context, req resource.SchemaRequest, resp *resource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		Attributes: map[string]schema.Attribute{
+			"app": schema.StringAttribute{
```
Comment (meetreks) [inline internal/service/appfabric/app_authorization.go:69]: Please consider adding a ENUM as it accepts only valid values.

File: internal/service/appfabric/app_authorization_test.go [hunk 3]
```diff
@@ -0,0 +1,437 @@
...
+	}
+}
+
+func testAccAppAuthorizationConfig_basic() string {
+	return `
+
+
+resource "aws_appfabric_app_authorization" "test" {
+  app_bundle_identifier = aws_appfabric_app_bundle.arn
+  app                   = "TERRAFORMCLOUD"
```
Comment (meetreks) [inline internal/service/appfabric/app_authorization_test.go:313]: Is this a valid value? As per this  doco - looks like itts noto a valid value -- https://docs.aws.amazon.com/appfabric/latest/api/API_CreateAppAuthorization.html

File: internal/service/appfabric/app_authorization_test.go [hunk 4]
```diff
@@ -0,0 +1,437 @@
...
+		return nil
+	}
+}
+
+func testAccAppAuthorizationConfig_basic() string {
+	return `
+
+
+resource "aws_appfabric_app_authorization" "test" {
+  app_bundle_identifier = aws_appfabric_app_bundle.arn
```
Comment (meetreks) [inline internal/service/appfabric/app_authorization_test.go:312]: Where is the resource for this?

File: internal/service/appfabric/app_authorization_test.go [hunk 5]
```diff
@@ -0,0 +1,437 @@
...
+			return err
+		}
+
+		*v = *output
+
+		return nil
+	}
+}
+
+func testAccAppAuthorizationConfig_basic() string {
```
Comment (meetreks) [inline internal/service/appfabric/app_authorization_test.go:307]: The return requires formatting..

File: internal/service/appfabric/appfabric_test.go [hunk 6]
```diff
@@ -0,0 +1,27 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
```
Comment (meetreks) [inline internal/service/appfabric/appfabric_test.go:3]: why is this file included?

File: website/docs/r/appfabric_app_authorization.html.markdown [hunk 7]
```diff
@@ -0,0 +1,80 @@
...
+Terraform resource for managing an AWS AppFabric App Authorization.
+
+## Example Usage
+
+### Basic Usage
+
+```terraform
+resource "aws_appfabric_app_authorization" "example" {
+  app                   = "TERRAFORMCLOUD"
+  app_bundle_identifier = aws_appfabric_app_bundle.arn
```
Comment (meetreks) [inline website/docs/r/appfabric_app_authorization.html.markdown:20]: include a resource for app_bundle.arn and the formatting is incorrect. You might want to consider aws_appfabric_app_bundle.test.arn

**Review by meetreks — COMMENTED** [review]
Excellent code, job well done. Just some minor edits to follow.

File: internal/service/appfabric/app_authorization.go (hunk 2 above)
Comment (markoskandylis) [reply internal/service/appfabric/app_authorization.go:69]: Yes true i will add the Enums for that :)

File: internal/service/appfabric/app_authorization_test.go (hunk 3 above)
Comment (markoskandylis) [reply internal/service/appfabric/app_authorization_test.go:313]: Yes its Valid the docs doenst  have all the possible names that was tested

File: internal/service/appfabric/app_authorization_test.go (hunk 4 above)
Comment (markoskandylis) [reply internal/service/appfabric/app_authorization_test.go:312]: This is still on creation the tests will be updated with the team that is working on that resource will make a pull request

File: internal/service/appfabric/appfabric_test.go (hunk 6 above)
Comment (markoskandylis) [reply internal/service/appfabric/appfabric_test.go:3]: This is a way to test the cases see example on Security lake, and bedrock knowledge base. @ewbankkit will that be the way forward for tests or we should default to the old tests

File: website/docs/r/appfabric_app_authorization.html.markdown (hunk 7 above)
Comment (markoskandylis) [reply website/docs/r/appfabric_app_authorization.html.markdown:20]: Yes the resource is not ready yet waiting for the pull request

**Review by ewbankkit — APPROVED** [review]
LGTM :rocket:.

```console
% AWS_DEFAULT_REGION=us-east-1 make testacc TESTARGS='-run=TestAccAppFabric_serial/AppAuthorization' PKG=appfabric 
make: Verifying source code with gofmt...
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go1.22.2 test ./internal/service/appfabric/... -v -count 1 -parallel 20  -run=TestAccAppFabric_serial/AppAuthorization -timeout 360m
=== RUN   TestAccAppFabric_serial
=== PAUSE TestAccAppFabric_serial
=== CONT  TestAccAppFabric_serial
=== RUN   TestAccAppFabric_serial/AppAuthorization
=== RUN   TestAccAppFabric_serial/AppAuthorization/apiKeyUpdate
=== RUN   TestAccAppFabric_serial/AppAuthorization/oath2Update
=== RUN   TestAccAppFabric_serial/AppAuthorization/tags
=== RUN   TestAccAppFabric_serial/AppAuthorization/basic
=== RUN   TestAccAppFabric_serial/AppAuthorization/disappears
--- PASS: TestAccAppFabric_serial (168.77s)
    --- PASS: TestAccAppFabric_serial/AppAuthorization (168.77s)
        --- PASS: TestAccAppFabric_serial/AppAuthorization/apiKeyUpdate (41.93s)
        --- PASS: TestAccAppFabric_serial/AppAuthorization/oath2Update (34.36s)
        --- PASS: TestAccAppFabric_serial/AppAuthorization/tags (46.55s)
        --- PASS: TestAccAppFabric_serial/AppAuthorization/basic (24.49s)
        --- PASS: TestAccAppFabric_serial/AppAuthorization/disappears (21.44s)
PASS
ok  	github.com/hashicorp/terraform-provider-aws/internal/service/appfabric	173.143s
```

**Comment by ewbankkit** [comment]
@markoskandylis Thanks for the contribution :tada: :clap:.
